### PR TITLE
Integration tests legalhold

### PR DESCRIPTION
**Pull request by kiran-chaudhary**
### Description of Change ###

Added integration tests for legalhold feature.

### PR Checklist ###
Did you remember to do the below?

- [ NA] Add unit tests to verify this change
- [ NA] Add an entry to CHANGELOG.md describing this change
- [ NA] Add docstrings for any new public parameters / methods / classes

### CODE REVIEW

**Comment by kiran-chaudhary** [comment]
Created draft PR because of failing tests.

File: integration/test_legalhold.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+user_uid = 977335752891390447
+policy_uid = 894163190167359975
+timestamp = str(int(datetime.now().timestamp()))
+matter_name = "integration test matter {}".format(timestamp)
+policy_name = "integration test policy {}".format(timestamp)
+
+
+@pytest.fixture(scope='module')
+def test_create_policy(connection):
```
Comment (kiran-chaudhary) [inline integration/test_legalhold.py:12]: Should this be renamed to `legalholduid`? Needed this to make sure a new policy is created for every test execution. 

Had to create it as a fixture to get the new id in other methods. However, pytest doesn't consider a fixture as a test.

File: integration/test_legalhold.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+user_uid = 977335752891390447
+policy_uid = 894163190167359978941631901673599755
+timestamp = str(int(datetime.now().timestamp()))
+matter_name = "integration test matter {}".format(timestamp)
+policy_name = "integration test policy {}".format(timestamp)
+
+
+@pytest.fixture(scope="module")
+def policy(connection):
```
Comment (kiran-chaudhary) [inline integration/test_legalhold.py:13]: This covers the create_policy test case.

**Comment by kiran-chaudhary** [comment]
added some test cases as fixtures, so that we can use the return values in other tests.

File: integration/test_legalhold.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+def test_get_all_matters(connection):
+    response_gen = connection.legalhold.get_all_matters()
+    for response in response_gen:
+        assert response.status_code == 200
+        break
+
+
+@pytest.mark.skip("Failing with 400 bad request error")
+def test_get_custodians_page(connection):
```
Comment (unparalleled-js) [inline integration/test_legalhold.py:68]: This is not a client method...

We should probably only be testing the client methods for integration tests

Comment (unparalleled-js) [inline integration/test_legalhold.py:68]: So this method needs either of one of it's parameters - it does do global search for some reason.

It needs one of these:
`legalHoldUid`
`userUid`
`user` (username)

**Review by unparalleled-js — CHANGES_REQUESTED** [review]
have to use 1 of the params for the failing test

File: integration/test_legalhold.py (hunk 3 above)
Comment (kiran-chaudhary) [reply integration/test_legalhold.py:68]: Fixed, can we make a note that necessary action can to be taken here. Either we document it or we fix the behaviour, or may be in version 2 we make one of them compulsory.

**Review by unparalleled-js — APPROVED** [review]

**Pull request merged**